Formula/fnm.rb

```diff
@@ -11,25 +11,37 @@ class Fnm < Formula
 
   bottle :unneeded
 
+  test do
+    system "#{bin}/fnm", '--version'
+  end
+
   def install
     bin.install 'fnm'
+
+    if configure_shell?
```

Comment:
Seems that the `caveats` method is being called more than once. And moving all side effects into `install` feels like it is the right thing to do, anyway.

Reply:
making `configure_shell?` memoized helps with knowing that we actually installed to the shell, and provide the right caveats at the end of the installation